tests/compression_tests.cpp

```diff
@@ -0,0 +1,147 @@
+
+
+#include <uv.h>
+#include "catch.hpp"
+#include "compression.h"
+
+
+TEST_CASE("deflate", "[http]") {
+
+    unsigned char packet_bytes[] = {
+  0x78, 0x9c, 0x3d, 0x8f, 0xbb, 0x0e, 0x82, 0x30,
+  0x14, 0x86, 0x77, 0x9e, 0xa2, 0xe9, 0x68, 0x6c,
+  0xa1, 0xca, 0xd5, 0xc4, 0x81, 0x81, 0xa8, 0x9b,
+  0x31, 0x98, 0xb8, 0x42, 0x7b, 0x2c, 0x4d, 0xa4,
+  0x25, 0xa5, 0x2e, 0x1a, 0xdf, 0xdd, 0x82, 0x86,
+  0xf1, 0xff, 0xfe, 0x4b, 0xce, 0x79, 0x07, 0x08,
+  0x61, 0x01, 0xf7, 0x47, 0xe3, 0x40, 0xe0, 0x1d,
+  0x72, 0xf6, 0x09, 0x6b, 0x34, 0xc1, 0x0e, 0x1a,
+  0x01, 0x76, 0xf4, 0xec, 0xed, 0xa5, 0x07, 0x25,
+  0xe7, 0x30, 0x38, 0xaf, 0xf1, 0x2a, 0x5c, 0xe1,
+  0x39, 0xb4, 0x50, 0x52, 0x69, 0x6e, 0x84, 0xd2,
+  0x72, 0xb2, 0xe5, 0x4b, 0x0d, 0x6b, 0xf4, 0x1f,
+  0x5d, 0x82, 0x47, 0x33, 0xce, 0xe5, 0xce, 0xb9,
+  0xa1, 0x55, 0x9a, 0x1a, 0x2b, 0x17, 0xef, 0x3a,
+  0x82, 0x25, 0xa5, 0x04, 0x3d, 0x27, 0x8e, 0x75,
+  0x7d, 0x56, 0x10, 0x32, 0x1a, 0xd1, 0xcd, 0x12,
+  0xb9, 0x91, 0xb2, 0x7f, 0x69, 0x52, 0xdb, 0x86,
+  0x03, 0x39, 0x4d, 0xa7, 0xe2, 0x8b, 0x31, 0x6e,
+  0xcf, 0x48, 0xba, 0x89, 0x58, 0x16, 0xdf, 0x33,
+  0x92, 0x25, 0x59, 0x91, 0xb2, 0x96, 0x25, 0x79,
+  0xd4, 0x02, 0x08, 0x91, 0xb4, 0xa9, 0x60, 0xc0,
+  0x0b, 0xec, 0x17, 0x3e, 0xbf, 0xa7, 0x7a, 0x70,
+  0x9d, 0x99, 0xcb, 0x87, 0xaa, 0xfe, 0x6d, 0x63,
+  0x63, 0x95, 0x54, 0x7a, 0x62, 0x39, 0x2d, 0x68,
+  0x1e, 0x53, 0x16, 0x6f, 0x71, 0xf0, 0x09, 0xbe,
+  0x24, 0x59, 0x49, 0xc2
+};
+
+    auto encodings = um_available_encoding();
+
+    struct res {
+        std::string str;
+        bool eof = false;
+    } res;
+    auto cb = [](void *ctx, const char *b, ssize_t len) {
+        auto out = (struct res *)ctx;
+        if (len > 0)
+            out->str.append(b, len);
+
+        if (len == UV_EOF) {
+            out->eof = true;
+        }
+    };
+
+    auto inflater = um_get_inflater("deflate", cb, &res);
+
+    um_inflate(inflater, (const char*)packet_bytes, sizeof(packet_bytes));
+
+    CHECK(um_inflate_state(inflater) == 1);
+    CHECK_THAT(res.str, Catch::Contains(R"("method": "GET")"));
```

Comment:
why not check the whole response is what you expect to see? worthwhile to check how many total bytes are returned to make sure it's correct?